src/euphorie/client/model.py

```diff
@@ -1486,6 +1507,77 @@ def in_place_custom_measures(self):
         return self.measures_of_type("in_place_custom")
 
 
+class Choice(SurveyTreeItem):
+    """User choice."""
+
+    __tablename__ = "choice"
+    __mapper_args__ = dict(polymorphic_identity="choice")
+
+    id = schema.Column(
+        "id",
+        types.Integer(),
+        schema.ForeignKey(SurveyTreeItem.id, onupdate="CASCADE", ondelete="CASCADE"),
+        primary_key=True,
+    )
+    condition = schema.Column(types.String(512), nullable=True)
+
+    @property
+    def is_visible(self):
+        if not self.condition:
+            return True
+        options = self.condition.split("|")
+        if (
+            Session.query(Option)
+            .join(Choice)
+            .filter(Choice.session_id == self.session_id)
+            .filter(Option.zodb_path.in_(options))
+            .count()
+            <= 0
+        ):
+            return False
+        return True
```

Comment:
This is not very readable.
Can count return less than 0?
```suggestion
        return (
            Session.query(Option)
            .join(Choice)
            .filter(Choice.session_id == self.session_id)
            .filter(Option.zodb_path.in_(options))
            .count()
        )
```
but maybe there is also a `.exists()`, I do not remember.